include/fastdds/rtps/writer/StatefulWriter.h

```diff
@@ -59,6 +59,12 @@ class StatefulWriter : public RTPSWriter
 
     void rebuild_status_after_load();
 
+    virtual void print_inconsistent_acknack(
+            const GUID_t& writer_guid,
+            const GUID_t& reader_guid,
+            const SequenceNumberSet_t& sn_set,
```

Comment:
Instead of using the complete set, I would use just pass the two sequence numbers. Something like `min_requested_sequence_number` and `max_requested_sequence_number`. This way we avoid the problems calculating max when the set is empty